src/routes/staking/staking.tsx

```diff
@@ -213,15 +214,9 @@ export const StakingStepSelectNode = ({
         userStakePercentage,
       };
     });
-    const sortedByStake = nodesWithPercentages.sort((a, b) => {
-      if (a.stakedOnNode.isLessThan(b.stakedOnNode)) return 1;
-      if (a.stakedOnNode.isGreaterThan(b.stakedOnNode)) return -1;
-      if (a.id < b.id) return 1;
-      if (a.id > b.id) return -1;
-      return 0;
-    });
+    const randomized = shuffle(nodesWithPercentages);
```

Comment:
This changes the order every single time we refresh the validators, which could lead to people clicking on the wrong node.

Reply:
@dexturr I've pushed quite a bit of new code here so worth having another look.